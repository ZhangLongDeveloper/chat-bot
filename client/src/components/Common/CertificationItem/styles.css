.certification-container .image {
  width: 100%;
  border-radius: 5px;
  box-shadow: 0px 0px 15px rgba(0,0,0,0.1);
  cursor: pointer;
}

.certification-container .image:hover {
  transform: scale(1.03);
  box-shadow: 0px 0px 25px rgba(0,0,0,0.2);
}

.certification-container .image-wrapper:hover + .title {
  transform: scale(1.05);
}

.certification-container .title {
  text-align: center;
  margin-top: 20px;
  font-size: 18px;
  font-weight: bold;
}

