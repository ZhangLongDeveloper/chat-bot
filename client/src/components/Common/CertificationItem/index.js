import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const CertificationItem = ({ title, image, url }) => (
  <div className="certification-container">
    <a href={url} className="image-wrapper" target="_blank" rel="noopener noreferrer">
      <img className="image" src={image} alt="certificate" />
    </a>
    <div className="title">{title}</div>
  </div>
);

CertificationItem.propTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default CertificationItem;
