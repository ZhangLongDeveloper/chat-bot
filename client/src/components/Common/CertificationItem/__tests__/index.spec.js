import React from 'react';
import { create } from 'react-test-renderer';

import CertificationItem from '../../CertificationItem';

const mockComponent = props => (
  <CertificationItem {...props} />
);

describe('components/Common/<CertificationItem />', () => {
  it('should render component with required props', () => {
    const props = {
      title: 'title',
      image: 'image',
      url: 'url',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
