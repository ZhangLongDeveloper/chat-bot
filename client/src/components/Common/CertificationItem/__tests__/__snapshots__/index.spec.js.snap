// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/<CertificationItem /> should render component with required props 1`] = `
<div
  className="certification-container"
>
  <a
    className="image-wrapper"
    href="url"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="certificate"
      className="image"
      src="image"
    />
  </a>
  <div
    className="title"
  >
    title
  </div>
</div>
`;
